src/main/drivers/bus_quadspi.h

```diff
@@ -0,0 +1,141 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Author: Dominic Clifton
+ */
+
+#pragma once
+
+#include "drivers/io_types.h"
+#include "drivers/rcc_types.h"
+
+#include "pg/pg.h"
+#include "pg/pg_ids.h"
+
+/*
+ * Quad SPI supports 1/2/4 wire modes
+ *
+ * 1LINE is like SPI MISO/MOSI using D0 (MOSI)/D1(MISO).
+ * 2LINE uses D0, D1 (bidirectional).
+ * 4LINE uses D0..D3 (bidirectional)
+ *
+ * See ST Micros' AN4760 "Quad-SPI (QSPI) interface on STM32 microcontrollers"
+ */
+
+#ifdef USE_QUADSPI
+
+#if defined(STM32H7)
+#define QUADSPI_IO_AF_BK_IO_CFG           IO_CONFIG(GPIO_MODE_AF_PP, GPIO_SPEED_FREQ_VERY_HIGH, GPIO_NOPULL)
+#define QUADSPI_IO_AF_CLK_CFG             IO_CONFIG(GPIO_MODE_AF_PP, GPIO_SPEED_FREQ_VERY_HIGH, GPIO_NOPULL)
+#define QUADSPI_IO_AF_BK_CS_CFG           IO_CONFIG(GPIO_MODE_AF_PP, GPIO_SPEED_FREQ_VERY_HIGH, GPIO_PULLUP)
+#define QUADSPI_IO_BK_CS_CFG              IO_CONFIG(GPIO_MODE_OUTPUT_PP, GPIO_SPEED_FREQ_HIGH, GPIO_PULLUP)
+#else
+#error Quad SPI unsupported on this MCU
```

Comment:
Are there plans to add quadSPI support for other MCU types in the near future? If not, we can just do 

```
#if !defined(STM32H7)
#error Quad SPI unsupported on this MCU
#endif
```

and not bother with STM32H7 conditionals in the rest of the code for now.

Reply:
May be F7? Not in the "near" future, though.

Reply:
Since this is clearly not building on any platform except STM32H7 at the moment there is no way that we can assert that the `#if defined(STM32H7)` conditionals are anywhere near being correct. This means that they will complicate ongoing maintenance of the code, but it will be impossible to test if a change does not break the non-STM32H7 case.
For these reasons I am for pulling the defines out of the code for now - we'll have to go over what is needed and what not at the point that another platform is added anyway.

Reply:
I'm convinced. Will make the change.